### DISCUSSION TITLE
410 error

### DISCUSSION THREAD

**Discussion by Rit76Horizon** [streamlink/streamlink#4128] [Q&A] [answered]
I'm getting a 410 error when trying to use Streamlink, I updated to the latest release today as I had not used the program in some time, since my primary use for the program is to record unarchived streams, One of which is happening later today.

I thought I'd try and use it on a stream to make sure everything was still working, noticed I was still on 2.0.0, and updated. Now that I've updated I get a 410 error every time I try and view/record a stream. What can I do to fix this and what information is needed to resolve this.

**Comment by beardypig** [comment, marked as answer]
Can you provide the URL that you are trying to access and a copy of the output from streamlink when using the `-l debug` option?

**Reply by Rit76Horizon** [reply]
Since the streams for people I follow are over right now, here's the one I just tested with https://www.youtube.com/watch?v=VDWxsqu6e1Y 

It says -1 debug is an unrecognized argument, so for now this is what it shows if I just input things as basic as possible, which is what I've always done for the most part prior to this.

```
streamlink https://www.youtube.com/watch?v=VDWxsqu6e1Y best
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/watch?v=VDWxsqu6e1Y
error: Unable to open URL: https://youtube.com/get_video_info (410 Client Error: Gone for url: https://www.youtube.com/get_video_info?video_id=VDWxsqu6e1Y&el=detailpage)
```

**Reply by bastimeyer** [reply]
> https://youtube.com/get_video_info

You are not using the latest version. This is a request which the youtube plugin of an older Streamlink version makes (lower than 2.2.0 - changed in #3797) in order to request the data from youtube.

Add the `--loglevel=debug` parameter to see which version you are using. Installing a different Streamlink version doesn't guarantee that you're actually using it.

Btw, discussions are not really meant for plugin issues. The issue template for plugin issues on the issue tracker asks you to provide a debug log output, so those kinds of questions can get solved immediately by the users themselves.

**Reply by Rit76Horizon** [reply]
I saw the issue about using an older version, that's why I uninstalled the version I had and installed the latest release here, or at least I assumed I did, because it said "2.4.0" on the zip file.

that command displays this 
`[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.8.6
[cli][debug] Streamlink: 2.0.0
[cli][debug] Requests(2.25.1), Socks(1.7.1), Websocket(0.57.0)`
So I guess I somehow didn't get the newest version despite that being what I supposedly downloaded.

I didn't know this was a plugin issue, I'm not knowledgable in this area, so my natural assumption was to take it to the place meant for things that aren't plug-in issues or bug reports. 

Not sure what I'm meant to do as a lot of the stuff in the FAQ and guides, and so on doesn't make a much sense to me thanks to my lack of familiarity on these topics.

**Reply by bastimeyer** [reply]
> because it said "2.4.0" on the zip file.

The zip file includes source code which is meant for people installing it in their python environment. This is not what you want as a regular end-user on Windows. You need to download the installer.
https://streamlink.github.io/install.html

Make sure to fully uninstall previous installs of Streamlink.

**Reply by Rit76Horizon** [reply]
I don't understand, From the page you linked I went through " the Windows Binaries" section next to installers, from there it brings me to the 2.4.0 release here, Am I meant to download 2.4.0.exe from the assets section? For context what I did prior to this was go to the main page where it says code and has a zip download, This is what I've done for everything I've gotten from Github, and the previous version I had used. I'm going to assume this is incorrect ?

Was using the Uninstaller from the version I had not enough to actually uninstall the previous version? What steps do I take to make sure it's completely gone before I proceed? I know these are probably simple things that I'm struggling with so I apologize for that.

**Reply by beardypig** [reply]
@Rit76Horizon unless you have a special reason not to do so, you should indeed use the `exe` installer from the releases page.
